indexer/workers/importer.py

```diff
@@ -150,32 +153,38 @@ def process_story(
                     logger.error(f"Value for key '{key}' is not provided.")
                     continue
 
-            url = content_metadata.get("url")
-            assert isinstance(url, str)
-            url_hash = hashlib.sha256(url.encode("utf-8")).hexdigest()
             keys_to_skip = ["is_homepage", "is_shortened"]
             data: Mapping[str, Optional[Union[str, bool]]] = {
                 k: v for k, v in content_metadata.items() if k not in keys_to_skip
             }
-            self.import_story(url_hash, data)
+            self.import_story(data)
 
     def import_story(
         self,
-        url_hash: str,
         data: Mapping[str, Optional[Union[str, bool]]],
     ) -> Optional[ObjectApiResponse[Any]]:
         """
         Import a single story to Elasticsearch
         """
         response = None
         if data:
+            url = str(data.get("url"))
+            url_hash = hashlib.sha256(url.encode("utf-8")).hexdigest()
             publication_date = str(data.get("publication_date"))
+            # Add the indexed_date with today's date in ISO 8601 format
+            indexed_date: str = datetime.now().isoformat()
+            data = {**data, "indexed_date": indexed_date}
             target_index = self.index_routing(publication_date)
             try:
                 response = self.connector.index(url_hash, target_index, data)
-            except Exception as e:
-                response = None
-                logger.error(f"Elasticsearch exception: {str(e)}")
+            except ConflictError as e:
+                self.incr(
+                    "imported-stories", labels=[("status", "409-duplicate-stories")]
+                )
+                raise QuarantineException("%s", str(e))
+            except RequestError as e:
+                self.incr("imported-stories", labels=[("status", "400-indexing-error")])
+                raise QuarantineException("%s", str(e))
```

Comment:
```suggestion
            except ConflictError as e:
                self.incr(
                    "imported-stories", labels=[("status", "409-duplicate-stories")]
                )
                raise QuarantineException from e
            except RequestError as e:
                self.incr("imported-stories", labels=[("status", "400-indexing-error")])
                raise QuarantineException from e
```
I think [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining) is better than casting if I understood the code.

Reply:
@kilemensi -- I didn't write the QuarantineException catcher with chained exceptions in mind, so don't know how the code at https://github.com/mediacloud/story-indexer/blob/main/indexer/worker.py#L550 will handle your suggestion.

My comments on `QuarantineException("%s", str(e))` as submitted:
1. logger functions do % formatting, but QuarantineException is a class.
2. str(ExceptionSubclass("foo")) returns "foo", which is less useful than repr(ExceptionSubclass("foo")) which returns 'ExceptionSubclass("foo")', which gives the class name.

Not related to the exceptions, but in the above highlighted code is that I think the "imported-stories" counter should be changed to "stories", since the error events are stories that are NOT imported.

There usually isn't much room in the grafana graph windows (legend items are frequently not shown in full), so I try to keep names, label names and label values short, so I recommend using "dup" instead of "409-duplicate-stories".   Duplicates will happen regularly, so I don't think we want to collect them in quarantine.  We could start by doing it, but I think we would turn it off quickly.

For "400-indexing-error", I think "reqerr" is a reasonable short version of "RequestError", and if we don't understand all the ways this can happen, then collecting the stories in quarantine is a good idea.